liblttng-ust-java-agent/java/lttng-ust-agent-common/org/lttng/ust/agent/AbstractLttngAgent.java

```diff
@@ -324,31 +280,43 @@ public boolean appContextDisabled(String contextRetrieverName, String contextNam
 
 	@Override
 	public boolean isEventEnabled(String eventName) {
-		/* If at least one session enabled the "*" wildcard, send the event */
-		if (enabledWildcards.get() > 0) {
-			return true;
+		Boolean cachedEnabled = enabledEventNamesCache.get(eventName);
```

Comment:
Indeed, since there are non-protected accesses this map should be the concurrent one now.